src/fidesops/task/filter_element_match.py

```diff
@@ -85,7 +109,10 @@ def _remove_paths_from_row(
         # Loop through array in reverse to delete indices
         for i, _ in reversed(list(enumerate(matched_array))):
             if i not in preserve_index_list:
```

Comment:
We might want a comment here explaining that "this is referring to which index in the list to preserve, not which elements in the dataset to preserve" since in the erasures, any items preserved in this list will be masked (if I've understood that correctly?)

Reply:
Correct, it's which index we want to preserve, I will try to improve clarity.  Maybe I shouldn't even be talking about access/erasure here in this method at all, this should be a standalone helper method that is just removing/replacing elements in arrays as part of a post-processing step.

Importantly, this is also used for access requests too, basically if it's "preserved" here, it's data we care about, it's data we may return in an access request, or mask in an erasure request.

Reply:
That this method would return `row` in two different contexts feels like a potential foot-gun to me.

To confirm my understanding:
- in the context of an access request, it's fine to remove the unwanted elements from `row`, because the data is being returned, and we don't need to preserve list order
- in the context of an erasure request, we'll need to go back and mask over these elements in the datastore, so we need to preserve the indexes in the list in order to effectively erase the data at those indexes

Especially since we're storing the results separately in either case (placeholder vs. regular), does this mean we're caching a lot of unnecessary PII considering we may only ever need one of those versions?

Reply:
> To confirm my understanding:

> - in the context of an access request, it's fine to remove the unwanted elements from row, because the data is being returned, and we don't need to preserve list order
> - in the context of an erasure request, we'll need to go back and mask over these elements in the datastore, so we need to preserve the indexes in the list in order to effectively erase the data at those indexes




Correct. In addition, on the access request, we remove the unwanted elements from the array because data in that array is potentially used to locate data on other collections downstream in the graph.

Reply:
Can you clarify what you mean by "we may only ever need one of those versions?"  as in, the policy only has access rules defined or only has erasure rules defined?

Reply:
updates - I've tried to clarify in code comments that this is a list of indices we want to preserve

Reply:
> Can you clarify what you mean by "we may only ever need one of those versions?" as in, the policy only has access rules defined or only has erasure rules defined?

When I wrote that I was under the impression it is the case that erasures will only ever use the placeholder data. As I understand it now, the access data may well be used for further lookups down the chain. So we do in fact need both